scripts/release_checks.sh

```diff
@@ -236,29 +242,35 @@ verify_prime_registry () {
         fi
     
         RKE2_RUNTIME_COUNT=$(wc -l < "${RKE2_RUNTIME_OUTFILE}")
-        verify_count "${RKE2_RUNTIME_COUNT}" "4" "RKE2 Runtime (in prime registry)"
-        rm -rf "${RKE2_RUNTIME_OUTFILE}"
+        verify_count "${RKE2_RUNTIME_COUNT}" "1" "RKE2 Runtime(in prime registry)"
     fi
 
-    # Verify system-agent-installer and upgrade images in prime registry for k3s and rke2 products
-    URL_ITEMS="system-agent-installer-${PRODUCT} ${PRODUCT}-upgrade"
+    if echo "${VERSION_PREFIX}" | grep -q "rc"; then
+        SYS_AGENT_INSTALLER_URL="docker://stgregistry.suse.com/rancher/system-agent-installer-${PRODUCT}"
+    else
+        SYS_AGENT_INSTALLER_URL="docker://registry.rancher.com/rancher/system-agent-installer-${PRODUCT}"
+    fi
+    debug_log "skopeo list-tags ${SYS_AGENT_INSTALLER_URL} | grep ${VERSION_PREFIX} | grep ${VERSION_SUFFIX} | tee -a ${SYS_AGENT_OUTFILE}"
+    skopeo list-tags "${SYS_AGENT_INSTALLER_URL}" | grep "${VERSION_PREFIX}" | grep "${VERSION_SUFFIX}" | tee -a "${SYS_AGENT_OUTFILE}"
+    
+    SYS_AGENT_COUNT=$(wc -l < "${SYS_AGENT_OUTFILE}")
+    verify_count "${SYS_AGENT_COUNT}" "1" "System Agent Installer for ${PRODUCT} (in prime registry)"
 
-    for ITEM in $URL_ITEMS; do
-        PRIME_URL="docker://registry.rancher.com/rancher/${ITEM}"
-        OUTFILE="${ITEM}_${VERSION_PREFIX}_${RANDOM_INT}"
+    if echo "${VERSION_PREFIX}" | grep -q "rc"; then
+        UPGRADE_INSTALLER_URL="docker://stgregistry.suse.com/rancher/${PRODUCT}-upgrade"
```

Comment:
One suggestion is to make this variable and if else once so that we reduce the duplication of checking if `${VERSION_PREFIX}` has an rc... like having a func like `set_url` and set the url based on that... cc @aganesh-suse

Reply:
```
set_url() {
  if echo "${VERSION_PREFIX}" | grep -q "rc"; then 
    URL="docker://stgregistry.suse.com/rancher"
  else
    URL="docker://registry.rancher.com/rancher"
}
```

and then the downstream vars can be
```
UPGRADE_INSTALLER_URL="${URL}/${PRODUCT}-upgrade"
```